### PR TITLE
PAPP-4148: Updating custom view for who is oncall

### PR DESCRIPTION
**Pull request by ebhoir-crest**
### This MR change contains
- Updating table view with the custom view for the new action get oncall user
- Updating JSON file, fixed example values, and datapath related issues
- Updating copyright year
- Updating release notes for FIPS compliant

### CODE REVIEW

**Review by pdesai-crest — DISMISSED** [review]

File: xmatters.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
     "type": "devops",
     "publisher": "Splunk",
     "main_module": "xmatters_connector.py",
-    "app_version": "2.1.1",
-    "utctime_updated": "2021-12-30T02:32:46.000000Z",
+    "app_version": "2.1.0",
```
Comment (ngaglani-crest) [inline xmatters.json:8]: `app_version` should be 2.1.1. `git pull origin next` should fix this issue.

File: xmatters_get_oncall_user.html [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+either express or implied. See the License for the specific language governing permissions
+and limitations under the License.
+-->
+
+<style>
+
+
+    .threatstream a:hover {
```
Comment (ngaglani-crest) [inline xmatters_get_oncall_user.html:30]: Please change the class name to `.xmatters` wherever applicable.

File: xmatters_get_oncall_user.html [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    .threatstream .glyphicon.glyphicon-dot:before {
+        content: "\25cf"; font-size: 10px;
+    }
+
+    .threatstream a.nowrap {
+        white-space:nowrap;
+    }
+
+</style>
+<div class="threatstream" style="overflow: auto; width: 100%; height: 100%; padding-left:10px; padding-right:10px"> <!-- Main Div -->
```
Comment (ngaglani-crest) [inline xmatters_get_oncall_user.html:93]: Same as above.

**Review by ngaglani-crest — APPROVED** [review]

File: xmatters.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "type": "devops",
     "publisher": "Splunk",
     "main_module": "xmatters_connector.py",
-    "app_version": "2.1.0",
+    "app_version": "2.1.1",
     "utctime_updated": "2021-07-23T07:02:13.000000Z",
```
Comment (ngaglani-crest) [inline xmatters.json:9]: `utctime_updated ` should also be updated.

**Review by ngaglani-crest — DISMISSED** [review]

**Review by alexa-phantom — APPROVED** [review]

File: xmatters_connector.py [hunk 5]
```diff
@@ -644,13 +648,24 @@ def _get_oncall_user(self, param):
 
         for res in response_json['data']:
             action_result.add_data(res)
+        i = 0
+        while i <= (len(response_json['data']) - 1):
+            for k, j in (response_json['data'][i]).items():
+                if k == 'shift':
+                    shift_count += 1
+            i += 1
```
Comment (paulnawat) [inline xmatters_connector.py:656]: suggestion: what do you think about using `enumerate` instead? It should simplify this block:

image

**Review by paulnawat — APPROVED** [review]
lgtm. just left one minor suggestion.

File: xmatters_connector.py (hunk 5 above)
Comment (ebhoir-crest) [reply xmatters_connector.py]: improved the code for shift count

**Comment by dgopani-crest** [comment]
Merging with 2 approvals

**Pull request merged**